src/gen_oauth2.erl

```diff
@@ -0,0 +1,178 @@
+-module(gen_oauth2).
```

Comment:
Would be good to have a header here that describes the two oauth flows: initiating a new handshake for a new user or supplying a previously stored access token to setup a client.